aten/src/ATen/native/cuda/IGammacKernel.cu

```diff
@@ -0,0 +1,27 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/cuda/Loops.cuh>
+#include <ATen/native/cuda/Math.cuh>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/BinaryOps.h>
+
+// NOTE: CUDA on Windows requires that the enclosing function
+// of a __device__ lambda not have internal linkage.
+
+namespace at { namespace native {
+
+void igammac_kernel_cuda(TensorIterator& iter) {
+
+  AT_DISPATCH_FLOATING_TYPES(iter.common_dtype(), "igammac_cuda", [&]() {

```

Comment:
Why doesn't CUDA promote half and bfloat16 to float for its computation, too?

Reply:
It is the same reason with `igamma`: it fails the test with bfloat16 and half, even though it's promoted to float32 internally.